docs/src/main/asciidoc/vault.adoc

```diff
@@ -443,6 +443,33 @@ The last relevant property is `quarkus.vault.tls.skip-verify`, which allows to c
 but without checking the certificate authenticity. This may be convenient in development, but is strongly
 discouraged in production as it is not more secure than talking to Vault in plain HTTP.
 
+== Vault Provisioning
+
+Beside the typical client use cases, the Quarkus extension can be used to provision Vault as well, 
+for instance as part of a CD pipeline. Specific endpoints support this scenario:
```

Comment:
Are they endpoints or CDI beans?

I think we should be clearer about how a user can make use of them.

Reply:
hello. I have made it more explicit, plus I added an example.